specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,318 @@
+                            ::ShellExecuteEx(&info);
+                        }
+                        else
+                        {
+                            // To display a custom dialog we cancel the launch, display
+                            // a custom dialog, and then manually launch the external protocol. 
+                            wil::unique_cotaskmem_string initiating_origin;
+                            CHECK_FAILURE(args->get_InitiatingOrigin(&initiating_origin));
+                            std::wstring message = L"Launching External Protocol request";
+                            if (initiating_origin.get() == L"")
+                            {
+                                message += L"from ";
+                                message += initiating_origin.get();
+                            }
+                            message += L" to ";
+                            message += uri.get();
+                            message += L"?\n\n";
+                            message += L"Do you want to grant permission?\n";
+                            int response = MessageBox(
+                                nullptr, message.c_str(), L"Launching External Protocol",
+                                MB_YESNOCANCEL | MB_ICONWARNING);
+                            if (response == IDYES)
+                            {
+                                std::wstring protocol_url = uri.get();
+                                SHELLEXECUTEINFO info = {sizeof(info)};
+                                info.fMask = SEE_MASK_NOASYNC;
+                                info.lpVerb = L"open";
+                                info.lpFile = protocol_url.c_str();
+                                info.nShow = SW_SHOWNORMAL;
+                                ::ShellExecuteEx(&info);
+                            }
+                        }
+                        return S_OK;
+                    };
+                    wil::com_ptr<ICoreWebView2Deferral> deferral;
+                    CHECK_FAILURE(args->GetDeferral(&deferral));
+
+                    m_appWindow->RunAsync(
+                        [deferral, showDialog]()
+                        {
+                            showDialog();
+                            CHECK_FAILURE(deferral->Complete());
+                        });
+                    return S_OK;
+                })
+                .Get(),
+            &m_launchingExternalProtocolToken));
+    }
+}
+    
+```
+
+## .NET and WinRT
+
+```c#
+private WebView2 webView;
+void RegisterLaunchingExternalProtocolHandler() 
+{
+    webView.CoreWebView2.LaunchingExternalProtocol += (sender, args) {
+        {
+            CoreWebView2Deferral deferral = args.GetDeferral();
+            System.Threading.SynchronizationContext.Current.Post((_) =>
+            {
+                using (deferral)
+                {
+                    // Set the `Cancel` property to `TRUE`, as we will either silently launch the 

```

Comment:
Is this a recommended best practice to cancel all launches if you're going to handle any of them?  I expect it would be more common for a page to "handle" (Cancel) some URI Schemes but let the browser default continue to handle most of them.

Reply:
Maybe a better sample would be

* calculator:// always allow
* malicious:// always block
* contoso:// custom prompt
* others: Fall back to system dialog

Reply:
Yes please update sample to show all the different cases. With comments to explain each case and why we're handling it that way.